code/datums/state_machine/paper_fortune_fsm.dm

```diff
@@ -0,0 +1,73 @@
+/decl/state/paper_fortune
+	var/state_name
+	var/list/fortune_choices
+	var/icon_state = "closed"
+
+/decl/state/paper_fortune/entered_state(obj/item/paper_fortune_teller/P)
+	if(istype(P))
+		P.update_icon()
+
+// The starting point.
+/decl/state/paper_fortune/closed
+	transitions = list(/decl/state_transition/paper_fortune/closed_to_open_vertical)
+	state_name = "closed"
+	fortune_choices = list(
+		"Red" = "#ff8888",
+		"Green" = "#88ff88",
+		"Blue" = "#8888ff",
+		"Yellow" = "#ffff88"
+	)
+
+/decl/state/paper_fortune/closed/entered_state(obj/item/paper_fortune_teller/P)
+	..()
+	if(istype(P))
+		P.choice_counter = 0
+
+// Alternates between this and the below state.
+/decl/state/paper_fortune/open_vertical
+	transitions = list(
+		/decl/state_transition/paper_fortune/vertical_to_horizontal,
+		/decl/state_transition/paper_fortune/open_to_closed
```

Comment:
OK, actually I guess you can unit test: for every FSM, start with initial state, then poputate a list of all states accessible from there via `transitions`. For each state, the FSM's expected type shall be a subtype of the expected type of the state.

Would need to add expected type to each state, of course.